core/src/main/java/ai/timefold/solver/core/impl/score/stream/bavet/common/index/ComparisonIndexer.java

```diff
@@ -43,8 +43,8 @@ public ComparisonIndexer(JoinerType comparisonJoinerType, int propertyIndex,
     }
 
     @Override
-    public ElementAwareListEntry<T> put(IndexProperties indexProperties, T tuple) {
-        Key_ indexKey = indexProperties.toKey(propertyIndex);
+    public ElementAwareListEntry<T> put(Object indexProperties, T tuple) {
```

Comment:
Can we guarantee that the type of `indexProperties` is always `Key_`?

It seems better to me that using a contract requiring the type `Key_` is preferable to allowing two possible types.

Reply:
Actually, `Key_` is here just so that the code is a bit shorter.
Technically, when generics is erases, it becomes `Comparable` - and that is as much as we know at this point. We never head the information about the type of the key - only the user knows that.